api/java/org/openyolo/api/CredentialClient.java

```diff
@@ -222,6 +250,63 @@ public Credential getCredentialFromActivityResult(@Nullable Intent resultData) {
         return responders;
     }
 
+    @Nullable
+    private ComponentName getPreferredProvider(@NonNull List<ComponentName> providers) {
```

Comment:
Very good point. Amended the heuristic such that if there are any unknown providers installed, there is no preferred provider.